drivers/serial/Kconfig.nrfx

```
@@ -50,6 +50,13 @@ config UART_NRF_DK_SERIAL_WORKAROUND_WAIT_MS
 	default 7
 	help
 	  Busy wait time (in milliseconds).
+
+# Increase stack size in tests built with no optimizations, as with the above
+# workaround enabled, the standard size may be insufficient. Use 512 to align
+# with the FPU_SHARING case, where the default value is also modified.
+config TEST_EXTRA_STACK_SIZE
+	default 512 if TEST && NO_OPTIMIZATIONS
```

There is no guarantee that `Kconfig.nrfx` will be processed before `subsys/testsuite/Kconfig` (i.e. there is no guarantee that the default value here will be used instead of the default values defined elsewhere).

If any, this should go into a SoC or board defconfig, which is guaranteed to be processed before the non-defconfigs.

---

> There is no guarantee that `Kconfig.nrfx` will be processed before `subsys/testsuite/Kconfig` (i.e. there is no guarantee that the default value here will be used instead of the default values defined elsewhere).
> 

In this particular case, the order is ensured here:
https://github.com/zephyrproject-rtos/zephyr/blob/e195c3cad377f01ef0a5b376770ea70d8c189a06/Kconfig.zephyr#L45-L47

so drivers does come before `subsys/testsuite`.
but generally this is fragile to re-factoring of Kconfig code.

So I agree with @stephanosio that this may not be the optimal place for changing this default.

> If any, this should go into a SoC or board defconfig, which is guaranteed to be processed before the non-defconfigs.

or the arch.
We have several places where this is adjusted depending on arch:
https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/mips/Kconfig#L39
https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/sparc/Kconfig#L47
etc.

or SoC  / SoC series as pointed out by @stephanosio.

additional note, the `TEST &&` can be omitted as there is already a `depends on TEST` on the original symbol.
```suggestion
	default 512 if NO_OPTIMIZATIONS
```

---

Moved to boards' defconfig files.